### PR TITLE
Store the image-to-SOCI-index mapping as an ActionResult

### PR DESCRIPTION
**Pull request by iain-macdonald**
This PR changes the format of the image-to-SOCI-index that's stored in the ActionCache. Previously, a cache entry was written using the rehashed, salted image's config hash as the key and the serialized digest of the SOCI index as the value. However, this had a few drawbacks including increased difficulty debugging the ActionCache entries using the ActionCacheServer APIs and requiring multiple trips to the cache to unpack a SOCI index. This PR changes the image-to-index mapping to be stored as an ActionResult with all of the SOCI artifacts (index and ZToCs) stored as OutputFiles with a NodeProperty denoting their type. This should address both shortcomings mentioned previously.

**Related issues**: N/A

### CODE REVIEW

File: enterprise/server/sociartifactstore/sociartifactstore.go [hunk 1]
```diff
@@ -450,16 +436,27 @@ func (s *SociArtifactStore) indexImage(ctx context.Context, image v1.Image, conf
...
+	// - The digest of all artifacts (ZToCs and SOCI Index) as OutputFiles
+	var artifacts []*repb.OutputFile
+	artifacts = append(artifacts, &repb.OutputFile{Digest: indexResourceName.GetDigest()})
+	for _, ztocDigest := range ztocDigests {
+		artifacts = append(artifacts, &repb.OutputFile{Digest: ztocDigest})
+	}
+	req := repb.UpdateActionResultRequest{
+		ActionDigest: sociIndexCacheKey,
+		ActionResult: &repb.ActionResult{
+			StdoutRaw:   []byte(serializedIndexResourceName),
```
Comment (vadimberezniker) [inline enterprise/server/sociartifactstore/sociartifactstore.go:455]: I don't like overloading Stdout like this. What if we stored it as another output file, using path or node_properties to identify the type of data?

**Review by vadimberezniker — CHANGES_REQUESTED** [review]

File: enterprise/server/sociartifactstore/sociartifactstore.go (hunk 1 above)
Comment (iain-macdonald) [reply enterprise/server/sociartifactstore/sociartifactstore.go]: SG, done

File: enterprise/server/sociartifactstore/sociartifactstore.go [hunk 2]
```diff
@@ -285,59 +289,52 @@ func sociIndexKey(h v1.Hash) (*repb.Digest, error) {
...
+		} else if outputFile.NodeProperties.Properties[0].Value == ztocArtifactType {
+			ztocDigests = append(ztocDigests, outputFile.Digest)
+		} else {
+			recordOutcome("action_result_bad_node_prop_value")
+			return nil, status.InternalErrorf(
+				"malformed action result, unrecognized node property value: %s",
+				outputFile.NodeProperties.Properties[0].Value)
 		}
 	}
 	recordOutcome("cached")
```
Comment (vadimberezniker) [inline enterprise/server/sociartifactstore/sociartifactstore.go:336]: add a check here to verify that sociIndexDigest and ztocDigests were populated

**Review by vadimberezniker — APPROVED** [review]

File: enterprise/server/sociartifactstore/sociartifactstore.go (hunk 2 above)
Comment (iain-macdonald) [reply enterprise/server/sociartifactstore/sociartifactstore.go:340]: Added a check for the SOCI index. It's technically valid (but not very useful) for all the ZToCs to be missing if all of the image layers are below the minimum layer size, so I omitted that chekc.

**Review by vadimberezniker — APPROVED** [review]

**Pull request merged**